department-of-veterans-affairs/vets-website
# Add unassociated incomes

## THIS FORM IS DISABLED IN PRODUCTION

# Income and Asset Statement Form (21P-0969)

## Description
Add a new section with the [Array Builder Pattern(Multiple responses list & loop)](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/patterns/array-builder/README.md)

## Issue
[Section III: Recurring Income Not Associated with Accounts or Assets #86213](https://github.com/department-of-veterans-affairs/va.gov-team/issues/86213)

### Related Issues
n/a

### Epic
[Digitize Income and Asset Statement Form 21P-0969](https://app.zenhub.com/workspaces/benefits-pension-64e775aaa6b1ca1ed49b2ede/issues/gh/department-of-veterans-affairs/va.gov-team/80925)

### Screenshots
#### Intro page
| Pristine | Required |
| ------- |  ------- | 
|![localhost_3001_income-and-asset-statement-form-21p-0969_form-saved (3)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/129b5bfc-79a6-4c43-8cd0-cb05d3e0e1bb)|![localhost_3001_income-and-asset-statement-form-21p-0969_form-saved (4)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/65eead1d-8f92-4c90-a07c-d2d3f8875b20)|

#### Recurring income pages
| First page | Second page |
| ------- |  ------- |
|![localhost_3001_income-and-asset-statement-form-21p-0969_form-saved (5)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/9f634389-f460-43e4-bd43-52f044e9c694)|![localhost_3001_income-and-asset-statement-form-21p-0969_form-saved (6)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/e33d39bd-11ab-4318-96cb-fa28d5b9dae1)|

#### Summary page
| Single Entry | Multiple Entries |
| ------- |  ------- |
|![localhost_3001_income-and-asset-statement-form-21p-0969_form-saved (7)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/a37c2409-51d3-426c-b864-d084fe93cd0f)|![localhost_3001_income-and-asset-statement-form-21p-0969_form-saved (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/a3c446c9-b056-4a69-b869-2ecd586d32f3)|

#### Review and edit page
| Multiple Entries |
| ------- |
|![localhost_3001_income-and-asset-statement-form-21p-0969_form-saved (8)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/675800f5-6753-4433-b6c7-28d8c1ac3542)|

## How to run in local environment
1. Check out this branch locally in 'vets-website'
2.  Checkout this [branch](https://github.com/department-of-veterans-affairs/content-build/pull/2075) locally in 'content-build'
3. Run 'vets-api' in your local environment
4. Run the 'vets-website' app with `yarn watch --env entry=static-pages,auth,login-page,profile,dashboard,income-and-asset-statement`

## How to toggle feature flag in local environment
1. Go to [income_and_assets_form_enabled](http://localhost:3000/flipper/features/income_and_assets_form_enabled)

## How to verify
1. Go to `http://localhost:3001/income-and-asset-statement-form-21p-0969` in your browser
2. Sign in with a mock user and 'Start application'
3. Go to the 'Unassociated incomes' chapter
4. Verify that required fields are required
5. Verify that optional fields are optional
6. Verify that all fields follow validation rules
7. Verify that up to 5 unassociated incomes can be added, edited and deleted
8. Verify that the unassociated incomes are visible and editable on the 'Review and submit' page
9. Verify that the form can be saved via the api endpoint
10. Verify that the form can be submitted

## Testing

## What areas of the site does it impact?
Income and Asset Statement Application

### Quality Assurance & Testing

### Error Handling

### Authentication
